src/coap_client_zephyr.c

```diff
@@ -0,0 +1,1147 @@
+    int err;
+
+    err = golioth_coap_req_new(&coap_req, client, COAP_METHOD_GET, COAP_TYPE_CON,
+                               GOLIOTH_COAP_MAX_NON_PAYLOAD_LEN + path_len,
+                               golioth_coap_cb, req);
+    if (err) {
+        return err;
+    }
+
+    if (first) {
+        /* Save token for subsequent block requests */
+        client->block_token_len = coap_header_get_token(&coap_req->request, client->block_token);
+    } else {
+        /* Override tokeń with the one geenrated in first block request */
+        memcpy(&coap_req->request.data[4], client->block_token, client->block_token_len);
+    }
+
+    err = coap_packet_append_uri_path_from_pathv(&coap_req->request, pathv);
+    if (err) {
+        LOG_ERR("Unable add uri path to packet");
+        goto free_req;
+    }
+
+    coap_req->block_ctx.current = (req->get_block.block_index * req->get_block.block_size);
+
+    err = golioth_coap_req_append_block2_option(coap_req);
+    if (err) {
+        LOG_ERR("Unable to append block2: %d", err);
+        goto free_req;
+    }
+
+    err = golioth_coap_req_schedule(coap_req);
+    if (err) {
+        LOG_ERR("Failed to schedule CoAP GET BLOCK: %d", err);
+        goto free_req;
+    }
+
+    return 0;
+
+free_req:
+    golioth_coap_req_free(coap_req);
+
+    return err;
+}
+
+static int golioth_coap_observe(golioth_coap_request_msg_t* req, struct golioth_client* client) {
+    int err;
+
+    err = golioth_coap_req_cb(req->client, COAP_METHOD_GET,
+                              PATHV(req->path_prefix, req->path),
+                              golioth_content_type_to_coap_format(req->observe.content_type),
+                              NULL, 0,
+                              golioth_coap_cb, req,
+                              GOLIOTH_COAP_REQ_OBSERVE);
+    if (err) {
+        LOG_ERR("Failed to schedule CoAP OBSERVE: %d", err);
+        return err;
+    }
+
+    return 0;
+}
+
+static void reestablish_observations(struct golioth_client* client) {
+    golioth_coap_observe_info_t* obs_info = NULL;
+    for (int i = 0; i < CONFIG_GOLIOTH_MAX_NUM_OBSERVATIONS; i++) {
+        obs_info = &client->observations[i];
+        if (obs_info->in_use) {
+            golioth_coap_observe(&obs_info->req, client);
+        }
+    }
+}
+
+static enum golioth_status coap_io_loop_once(struct golioth_client* client) {
+    golioth_coap_request_msg_t* req;
+    enum golioth_status status = GOLIOTH_OK;
+    int err = 0;
+
+    req = calloc(1, sizeof(*req));
+    if (!req) {
+        status = GOLIOTH_ERR_MEM_ALLOC;
+        goto free_req;
+    }
+
+    // Wait for request message, with timeout
+    bool got_request_msg = golioth_mbox_recv(
+        client->request_queue, req, CONFIG_GOLIOTH_COAP_REQUEST_QUEUE_TIMEOUT_MS);
+    if (!got_request_msg) {
+        // No requests, so process other pending IO (e.g. observations)
+        LOG_DBG("Idle io process start");
+        /* coap_io_process(context, COAP_IO_NO_WAIT); */
```

Comment:
Is this still relevant? We should remove it if not

Reply:
No, it is not. Removed!